R/variablesChartsSubgroups.R

```diff
@@ -219,12 +217,37 @@ variablesChartsSubgroups <- function(jaspResults, dataset, options) {
     jaspGraphs::themeJaspRaw()
 
   if (manualXaxis != "") {
-    xLabels <- factor(manualXaxis, levels = manualXaxis)
-    p <- p + ggplot2::scale_x_continuous(breaks = 1:length(manualXaxis), labels = xLabels)
+    if (length(levels(manualXaxis)) == length(manualXaxis)){
+      xBreaks_Out <- unique(manualXaxis)
+      p <- p + ggplot2::scale_x_continuous(breaks = 1:length(manualXaxis), labels = levels(manualXaxis))
+    }
+    else{
+      data <- data1
```

Comment:
why don't you rename the original `data1` to `data`? Or just accept the weird name and use `data1` everywhere.